zebra-consensus/src/primitives/ed25519.rs

```diff
@@ -120,43 +126,32 @@ impl Verifier {
 
     /// Flush the batch using a thread pool, and return the result via the channel.
     /// This function returns a future that becomes ready when the batch is completed.
-    fn flush_spawning(batch: BatchVerifier, tx: Sender) -> impl Future<Output = ()> {
+    async fn flush_spawning(batch: BatchVerifier, tx: Sender) {
         // Correctness: Do CPU-intensive work on a dedicated thread, to avoid blocking other futures.
-        tokio::task::spawn_blocking(|| {
-            // TODO:
-            // - spawn batches so rayon executes them in FIFO order
-            //   possible implementation: return a closure in a Future,
-            //   then run it using scope_fifo() in the worker task,
-            //   limiting the number of concurrent batches to the number of rayon threads
-            rayon::scope_fifo(|s| s.spawn_fifo(|_s| Self::verify(batch, tx)))
-        })
-        .map(|join_result| join_result.expect("panic in ed25519 batch verifier"))
+        // TODO:
+        // - spawn batches so rayon executes them in FIFO order
+        //   possible implementation: return a closure in a Future,
+        //   then run it using scope_fifo() in the worker task,
+        //   limiting the number of concurrent batches to the number of rayon threads
+        let _ = tx.send(spawn_fifo(move || batch.verify(thread_rng())).await.ok());
     }
 
     /// Verify a single item using a thread pool, and return the result.
     /// This function returns a future that becomes ready when the item is completed.
-    fn verify_single_spawning(item: Item) -> impl Future<Output = VerifyResult> {
+    async fn verify_single_spawning(item: Item) -> Result<(), BoxError> {
         // Correctness: Do CPU-intensive work on a dedicated thread, to avoid blocking other futures.
-        tokio::task::spawn_blocking(|| {
-            // Rayon doesn't have a spawn function that returns a value,
-            // so we use a parallel iterator instead.
-            //
-            // TODO:
-            // - when a batch fails, spawn all its individual items into rayon using Vec::par_iter()
-            // - spawn fallback individual verifications so rayon executes them in FIFO order,
-            //   if possible
-            rayon::iter::once(item)
-                .map(|item| item.verify_single())
-                .collect()
-        })
-        .map(|join_result| join_result.expect("panic in ed25519 fallback verifier"))
+        // TODO:
+        // - when a batch fails, spawn all its individual items into rayon using Vec::par_iter()
+        // - spawn fallback individual verifications so rayon executes them in FIFO order,
```

Comment:
I think this TODO item has been achieved in this PR, and the other TODO is now redundant?
This applies to all of the verify_single_spawning() methods.

Reply:
Thank you, removed in https://github.com/ZcashFoundation/zebra/pull/6887/commits/02684ca1349a0c0513feecdba27a0dd5afb02a18.